src/scripts/create_estner.py

```diff
@@ -0,0 +1,101 @@
+# /// script
+# requires-python = ">=3.10,<4.0"
+# dependencies = [
+#     "datasets==4.0.0",
+#     "huggingface-hub==0.34.4",
+#     "requests==2.32.5",
+# ]
+# ///
+
+"""Create the Estonian NER dataset and upload to HF Hub."""
+from typing import MutableMapping
+
+from datasets import load_dataset, DatasetDict
+from huggingface_hub import HfApi
+from requests import HTTPError
+
+
+def main() -> None:
+    """Create the Estonian NER dataset and upload to HF Hub."""
+    target_repo_id = "EuroEval/estner-mini"
+
+    # start from the official source
+    ds = load_dataset("tartuNLP/EstNER")
+
+    # target split sizes
+    train_size = 1024
+    val_size = 256
+    test_size = 2048
+
+    ds = ds.shuffle(seed=42)
+
+    train_ds = ds["train"].select(range(train_size))
+    val_ds = ds["dev"].select(range(val_size))
+    test_ds = ds["test"].select(range(test_size))
+
+    ds = DatasetDict(
+        train=train_ds,
+        val=val_ds,
+        test=test_ds,
+    )
```

Comment:
Ruff (this should have happened during your pre-commit hook, not sure why it hasn't):


```suggestion
    ds = DatasetDict(train=train_ds, val=val_ds, test=test_ds)
```